File: 06 - Java/07-04-00-IntroducirDatosMatriz-ULP.java
import java.util.Scanner;

public class ProgramaPrincipal {

    public static void main(String[] args){
        // Definimos las variables
        int renglones, columnas;
        var consola = new Scanner(System.in);
        // Definimos matriz
        System.out.print("Proporciona los renglones: ");
        renglones = Integer.parseInt(consola.nextLine());
        System.out.print("Proporciona las columnas: ");
        columnas = Integer.parseInt(consola.nextLine());
        // Declaramos la matriz
        int[][] matriz = new int[renglones][columnas];
        // Solicitar los valores al usuario
        for(int ren=0; ren < renglones ; ren++){
            for(int col=0; col < columnas; col++){
                System.out.print("Dato[" + ren + "][" + col + "] = ");
                matriz[ren][col] = Integer.parseInt(consola.nextLine());
            }
        }
        // Iteramos la matriz
        System.out.println("-----Iteramos la Matriz ------");
        for(int ren = 0; ren < renglones; ren++ ){
            for(int col=0; col < columnas; col++ ){
                System.out.print("Matriz[" + ren + "][" + col + "]=" + matriz[ren][col] + " ");
            }
            System.out.println();
        }
    }
}

File: 03 - C++/07-04-00-IntroducirDatosMatriz-ULP.cpp
#include <iostream>
using namespace std;

int main() {
	// Definimos las variables
	int renglones, columnas;
	// Definimos matriz
	cout << "Proporciona los renglones: " ;
	cin >> renglones;
	cout << "Proporciona las columnas: ";
	cin >> columnas;
	
	// Declaramos la matriz
	//int matriz[renglones][columnas];
	int **matriz = new int*[renglones];
	for (int i = 0 ; i < renglones ; i++){
		matriz[i] = new int[columnas];
	}
	
	// solicitar los valores al usuario
	for(int ren=0; ren < renglones ; ren++){
		for(int col=0; col < columnas; col++){
			cout << "Dato[" << ren << "][" << col << "] = " ;
			cin >> matriz[ren][col];
		}
	}
	
	// Iteramos la matriz
	cout << "-----Iteramos la Matriz ------" << endl;
	for(int ren = 0; ren < renglones; ren++ ){
		for(int col=0; col < columnas; col++ ){
			cout << "Matriz[" << ren << "][" << col << "] = " << matriz[ren][col] << endl; 
		}
		cout << endl;
	}
	
	// Eliminamos la memoria solicitada
	for (int i = 0; i < renglones; i++){
		delete [] matriz[i];
	}
	delete [] matriz;
	
	return 0;
}

